core/templates/domain/skill/skill-creation-backend-api.service.ts

```diff
@@ -37,7 +37,9 @@ export interface SkillCreationBackendDict {
 
 export interface ImageData {
   filename: string;
-  imageBlob: Blob;
+  // 'imageBlob' will be null when filenames
+  // are not present in localStorage.
```

Comment:
I don't understand this comment, can you elaborate.

Reply:
Already used in many places. Means we are getting ImageBlob from localstorage. And there -> Image file returned will be null when blob is not of type  'image', blob size is zero or dataURI is null.  dataURI here will be null if imageFilename does not exist in local storage.

Comment is present everywhere(root level as well) why it is null.

Reply:
Sorry, still not 100% clear what you mean. "if imageFilename does not exist in local storage" do you mean that "the image does not exist in local storage"? Because storage contains images that are being named using a filename the filenames are not stored in the storage.

Reply:
how imageBlob is set to null.

![Screenshot from 2022-08-18 12-51-49](https://user-images.githubusercontent.com/80846488/185334867-90b7e73a-cb99-4f99-929a-42efb0607e44.png)

![Screenshot from 2022-08-18 12-52-21](https://user-images.githubusercontent.com/80846488/185334923-2a99894d-7739-44f0-a28d-757bcf11dc14.png)

Reply:
Ah I see now. Thinking about it though, I would move the `dataURI == null` check outside the `converImageDataToImageFile` and only call it when `dataURI` is not `null`.

Reply:
But there are two more cases in this function `if (blob.type.match('image') && blob.size > 0)` where we need null if they fails.

Reply:
@vojtechjelinek PTAL.